presto-druid/src/main/java/com/facebook/presto/druid/DruidQueryGenerator.java

```diff
@@ -220,9 +244,108 @@ public DruidQueryGeneratorContext visitTableScan(TableScanNode node, DruidQueryG
+                }
+            }
+
+            // now visit the child project node
+            DruidQueryGeneratorContext context = node.getSource().accept(this, contextIn.withVariablesInAggregation(variablesInAggregation));
+            requireNonNull(context, "context is null");
+            checkArgument(!node.getStep().isOutputPartial(), "Druid pushdown does not support partial aggregations");
+
+            // 2nd pass
+            Map<VariableReferenceExpression, Selection> newSelections = new LinkedHashMap<>();
+            Set<VariableReferenceExpression> groupByColumns = new LinkedHashSet<>();
+            Set<VariableReferenceExpression> hiddenColumnSet = new HashSet<>(context.getHiddenColumnSet());
+            int aggregations = 0;
+            boolean groupByExists = false;
+
+            for (DruidAggregationColumnNode expression : aggregationColumnNodes) {
+                switch (expression.getExpressionType()) {
+                    case GROUP_BY: {
+                        GroupByColumnNode groupByColumn = (GroupByColumnNode) expression;
+                        VariableReferenceExpression groupByInputColumn = getVariableReference(groupByColumn.getInputColumn());
+                        VariableReferenceExpression outputColumn = getVariableReference(groupByColumn.getOutputColumn());
+                        Selection druidColumn = requireNonNull(context.getSelections().get(groupByInputColumn), "Group By column " + groupByInputColumn + " doesn't exist in input " + context.getSelections());
+
+                        newSelections.put(outputColumn, new Selection(druidColumn.getDefinition(), druidColumn.getOrigin()));
+                        groupByColumns.add(outputColumn);
+                        groupByExists = true;
+                        break;
+                    }
+                    case AGGREGATE: {
+                        AggregationFunctionColumnNode aggregationNode = (AggregationFunctionColumnNode) expression;
+                        String druidAggregationFunction = handleAggregationFunction(aggregationNode.getCallExpression(), context.getSelections());
+                        newSelections.put(getVariableReference(aggregationNode.getOutputColumn()), new Selection(druidAggregationFunction, DERIVED));
+                        aggregations++;
+                        break;
+                    }
+                    default:
+                        throw new PrestoException(DRUID_PUSHDOWN_UNSUPPORTED_EXPRESSION, "Unsupported pushdown for Druid connector. Unknown aggregation expression:" + expression.getExpressionType());
+                }
+            }
+
+            // Handling non-aggregated group by
+            if (groupByExists && aggregations == 0) {
+                VariableReferenceExpression hidden = new VariableReferenceExpression(UUID.randomUUID().toString(), BigintType.BIGINT);
+                newSelections.put(hidden, new Selection("count(*)", DERIVED));
```

Comment:
Is `count` the only aggregation function in this case?

Reply:
yep, count(*) is the only case